keras_cv/models/stable_diffusion/stable_diffusion.py

```diff
@@ -235,6 +235,8 @@ def generate_image(
                 + unconditional_guidance_scale * (latent - unconditional_latent)
             )
             a_t, a_prev = alphas[index], alphas_prev[index]
+            if latent.dtype == "float16":
```

Comment:
I think we should be able to unconditionally cast latent to the dtype of latent_prev, e.g. `ops.cast(latent, latent_prev.dtype)`.

Is there a reason that won't work?

Reply:
Thank you very much for your comment.
I think you are right!
I will try your advice and modify the code.

Reply:
Thank you very much for your comment.
I think you are right!
I will try your advice and modify the code.

Reply:
@ianstenbit 
I tried your suggestion and modified the code!
Thank you!!
If there is no problem, please approve my modification.